### PR TITLE
Add exit 1 instead of panic on unwrap of null args

### PR DESCRIPTION
**Pull request by cogitantium**
Does not return the help-message on no-args but exits cleanly with code 1. I think this is sufficient as there's no obvious way to make clap emit its help-message.

### CODE REVIEW

**Review by NicEastvillage — APPROVED** [review]
Approved.
Note, I made some additional clean ups in #89, which still hasn't been reviewed.

**Review by jenrik — CHANGES_REQUESTED** [review]
https://docs.rs/clap/2.33.3/clap/struct.App.html#method.print_help

**Comment by NicEastvillage** [comment]
Note that the modified line has been deleted in PR #90. You might want to check that out and/or rebase from there.

**Comment by cogitantium** [comment]
@jenrik using https://docs.rs/clap/2.33.3/clap/struct.App.html#method.print_help is not a viable route as App is consumed upon using get_matches, making it difficult to call print_help after using the result of get_matches to check if subcommands exist.

@NicEastvillage and I tried to solve it by using the setting AppSettings::ArgRequiredElseHelp (https://docs.rs/clap/2.31.1/clap/enum.AppSettings.html#variant.ArgRequiredElseHelp), however there's issues with subcommands counting as arguments and if any default values are set (e.g. log-level), this effectively negates this setting.

The changes initially suggested is an easy fix and since this issue generates very little value for the time it takes, I think we should save it for later. I'll make an issue.

File: src/main.rs [hunk 1]
```diff
@@ -77,6 +77,32 @@ fn main_inner() -> Result<(), String> {
     setup_tracing(&args);
     trace!(?args, "commandline arguments");
 
+    let subargs = args.subcommand().1.unwrap_or(exit(1));
```
Comment (jenrik) [inline src/main.rs:80]: Check: Unreachable call

**Review by jenrik — CHANGES_REQUESTED** [review]
Something is fishy, `cargo check` complains that line 80 is unreachable

File: src/main.rs (hunk 1 above)
Comment (NicEastvillage) [reply src/main.rs:80]: Ah yes. `exit(1)` is NOT lazily evaluated, so the program stops here.

**Pull request closed without merging**